### PR TITLE
[TLS 1.3] further usage of std::span.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

**Review by reneme — COMMENTED** [review]
Thanks for taking this on! I think `std::span<>` bears a lot of potential, also for the library's public APIs.

File: src/lib/tls/tls13/tls_transcript_hash_13.h [hunk 1]
```diff
@@ -49,13 +50,7 @@ class BOTAN_TEST_API Transcript_Hash_State
...
       Transcript_Hash_State& operator=(Transcript_Hash_State&&) = default;
 
-      void update(const std::vector<uint8_t>& serialized_message)
-         {
-         update(serialized_message.data(), serialized_message.size());
-         }
-
-      // TODO: C++20 replace this C-style API with std::span
-      void update(const uint8_t* serialized_message, const size_t serialized_message_length);
+      void update(const std::span<const uint8_t> serialized_message_s);
```
[the remark this answers is not in the document]
Comment (devnexen) [reply src/lib/tls/tls13/tls_transcript_hash_13.h:53]: true I ve been a bit over-zealous here 🙂

File: src/tests/test_tls_transcript_hash_13.cpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ std::vector<Test::Result> transcript_hash()
       Botan_Tests::CHECK("update without an algorithm", [](Test::Result& result)
          {
          Transcript_Hash_State h;
-         result.test_no_throw("update is successful", [&] { h.update({0xba, 0xad, 0xbe, 0xef}); });
+         result.test_no_throw("update is successful", [&] { h.update(std::vector<uint8_t>{0xba, 0xad, 0xbe, 0xef}); });
```
[the remark this answers is not in the document]
Comment (devnexen) [reply src/tests/test_tls_transcript_hash_13.cpp:76]: oky doky..

**Review by reneme — APPROVED** [review]
Thanks! RISCV stage is failing, but it seems unrelated (package installation failed).

**Pull request merged**